lib/Dialect/FIRRTL/Transforms/LowerTypes.cpp

```diff
@@ -827,13 +812,15 @@ Value TypeLoweringVisitor::addArg(FModuleOp module, Type type,
   auto newValue = body->addArgument(type);
 
   // Save the name attribute for the new argument.
-  StringAttr nameAttr = getFIRRTLNameAttr(module.getArgAttrs(oldArgNumber));
-  if (nameAttr) {
-    auto newArgAttrs = getArgAttrs(nameAttr, nameSuffix, builder);
-    auto newArgNumber = newValue.getArgNumber() - originalNumModuleArgs;
-    auto newArgAttrName = getArgAttrName(newArgNumber);
-    newModuleAttrs.push_back(NamedAttribute(newArgAttrName, newArgAttrs));
-  }
+  StringAttr nameAttr =
+      getFIRRTLModuleArgNameAttr(module)[oldArgNumber].cast<StringAttr>();
+  Attribute newArg;
+  if (!nameAttr.getValue().empty())
+    newArg =
+        builder->getStringAttr(nameAttr.getValue().str() + nameSuffix.str());
+  else
+    newArg = builder->getStringAttr("");
```

Comment:
Does firrtl dialect actually allow empty port names?  If not, then all this can go away.  If not, we should check that in the verifier.